tcp_logger.go

```diff
@@ -75,6 +75,7 @@ func (t *TCPLogger) handleConnError(err error, b []byte) {
 		log.Println("Retry limit met on TCPLogger.. sleeping for a minute before continuing.")
 		t.logLines <- b
 		time.Sleep(t.sleepDuration)
+		t.RetryConnection()
```

Comment:
It might be good to log the error if it's not deadline exceeded or EOF.

Reply:
I like it.  I just added a log for anytime this function is invoked.